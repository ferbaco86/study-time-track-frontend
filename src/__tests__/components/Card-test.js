import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import Card from '../../components/Card';

describe('Rendering component', () => {
  it('creates a Card component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Card
              title="Test"
              time={5}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Test');
  });
});

describe('Card', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Card
            title="Test"
            time={5}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
