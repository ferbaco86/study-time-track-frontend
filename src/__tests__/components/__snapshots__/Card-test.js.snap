// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-bdfBwQ cOFkCz"
>
  <h4
    className="sc-gsTCUz ctLQUY"
  >
    Test
  </h4>
  <div
    className="sc-hKgILt BstyB"
  >
    <h2
      className="sc-dlfnbm jJeQ"
    >
      5
    </h2>
    <span>
      hr(s)
    </span>
  </div>
</div>
`;
